pkg/routeagent_driver/handlers/ovn/submarinerNonGWRouteHandler.go

```diff
@@ -0,0 +1,172 @@
+		k8sClientSet:    k8sClientset,
+	}
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) Init() error {
+	logger.Infof("Starting NonGatewayRouteHandler")
+
+	node, err := nodeutil.GetLocalNode(nonGatewayRouteHandler.k8sClientSet)
+	if err != nil {
+		return errors.Wrap(err, "error getting the g/w node")
+	}
+
+	annotations := node.GetAnnotations()
+
+	// TODO transitSwitchIP changes support needs to be added.
+	transitSwitchIP, ok := annotations[ovnTransitSwitchIPAnnotation]
+	if !ok {
+		logger.Infof("No transit switch IP configured")
+		return nil
+	}
+
+	nonGatewayRouteHandler.transitSwitchIP, err = jsonToIP(transitSwitchIP)
+	if !ok {
+		return errors.Wrapf(err, "error parsing the transit switch IP")
+	}
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) GetName() string {
+	return "submariner-nongw-route-handler"
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) GetNetworkPlugins() []string {
+	// TODO enable when we switch to new implementation
+	// return []string{cni.OVNKubernetes}
+	return []string{}
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) RemoteEndpointCreated(endpoint *submarinerv1.Endpoint) error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	nonGatewayRouteHandler.remoteEndpoints[endpoint.Name] = endpoint
+
+	if !nonGatewayRouteHandler.isGateway || nonGatewayRouteHandler.transitSwitchIP == "" {
+		return nil
+	}
+
+	_, err := nonGatewayRouteHandler.smClient.SubmarinerV1().
+		NonGatewayRoutes(endpoint.Namespace).Create(context.TODO(),
+		nonGatewayRouteHandler.newNonGatewayRoute(endpoint), metav1.CreateOptions{})
+	if !apierrors.IsAlreadyExists(err) {
+		return errors.Wrapf(err, "error processing the remote endpoint create event for %q", endpoint.Name)
+	}
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) RemoteEndpointRemoved(endpoint *submarinerv1.Endpoint) error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+	delete(nonGatewayRouteHandler.remoteEndpoints, endpoint.Name)
+
+	if !nonGatewayRouteHandler.isGateway || nonGatewayRouteHandler.transitSwitchIP == "" {
+		return nil
+	}
+
+	if err := nonGatewayRouteHandler.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace).Delete(context.TODO(),
+		endpoint.Name, metav1.DeleteOptions{}); err != nil {
+		return errors.Wrapf(err, "error deleting nonGatewayRoute %q", endpoint.Name)
+	}
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) TransitionToNonGateway() error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	nonGatewayRouteHandler.isGateway = false
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) TransitionToGateway() error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	nonGatewayRouteHandler.isGateway = true
+	if nonGatewayRouteHandler.transitSwitchIP != "" {
```

Comment:
Shortcut here to reduce indentation.

```suggestion
	if nonGatewayRouteHandler.transitSwitchIP == "" {
	    return nil
	}
```

Reply:
Done.